hooks/analytics/useInitAnalytics.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from "react";
+import { useRouter } from "next/router";
+import analytics from "lib/analytics";
+
+
+/**
+ * Initialize the analytic provider
+ * @returns {Object} useInitAnalytics
+ */
+export default function useInitAnalytics() {
+  const router = useRouter();
+
+  useEffect(() => {
+    const handleRouteChange = (url) => {
```

Comment:
Need to clarify: Does this track the first entry page aswell when this effect runs, or only subsequent route changes?
Might need to do a separate `analytics.page()` call at the beginning of the effect if that's the case.

Reply:
I did a couple of tests and it works for the first loading.